### PR TITLE
Add CI config and fix formatting and cleanup chip features

### PR DESCRIPTION
**Pull request by pawelchcki**
Hi I've copied the CI configuration from [stm32-rs/stm32g0xx-hal ](https://github.com/stm32-rs/stm32g0xx-hal) and updated it to use thumbv7em-none-eabihf target and test all chip features found in the source code.

+ reformated the source to make cargo fmt pass
+ reordered the features alphabetically
+ removed Cargo.lock

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -2,15 +2,24 @@
 #![allow(non_camel_case_types)]
 
 #[cfg(not(any(
-  feature = "stm32g431",
-  feature = "stm32g441",
-  feature = "stm32g491",
```
Comment (pawelchcki) [inline src/lib.rs:7]: had to remove stm32g491 as I couldn't find the feature in stm32g4 crate.

**Comment by pawelchcki** [comment]
Created test PR in my fork, all seems green.

https://github.com/unresc/stm32g4xx-hal/pull/2/checks

**Comment by pawelchcki** [comment]
Also pushed removal Cargo.lock since I think it shouldn't be needed for a library and it was annoying me a bit in my other branches :) 

@dotcypress Hey, sorry for pinging you directly again - @summner is my alt account I use for odd stuff, using my main here since I plan to push a few more patches.

I took the liberty to add the CI since its a nice to have safety net.

I know this crate is a bit abandoned, and don't want to ping you all the time :D - are there other people who have merge rights (but maybe don't have notifications set up ?)

Thanks in advance! :)

**Comment by dotcypress** [comment]
This is great, thank you!

**Pull request merged**